cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/RepositorySourceControlMetadataStore.java

```diff
@@ -58,14 +70,87 @@ private StructuredTable getRepositorySourceControlMetadataTable() {
     return repositorySourceControlMetadataTable;
   }
 
+  /**
+   * Scans the source control metadata records based on the specified request parameters, and
+   * invokes the provided consumer for each matched record.
+   *
+   * @param request  The {@link ScanSourceControlMetadataRequest} object specifying the criteria for
+   *                 scanning the metadata records.
+   * @param type     The type of metadata records to scan. Currently, it is
+   *                 {@code APPLICATIONREFERENCE} by default.
+   * @param consumer The consumer to be invoked for each matched metadata record.
+   * @return The number of metadata records scanned and processed.
+   * @throws IOException If an I/O error occurs while scanning the metadata records.
+   */
+  public int scan(ScanSourceControlMetadataRequest request,
+      String type, Consumer<SourceControlMetadataRecord> consumer) throws IOException {
+    Range.Bound startBound = Range.Bound.INCLUSIVE;
+    final Range.Bound endBound = Range.Bound.INCLUSIVE;
+    Collection<Field<?>> startFields = new ArrayList<>();
+
+    startFields.add(
+        Fields.stringField(StoreDefinition.RepositorySourceControlMetadataStore.NAMESPACE_FIELD,
+            request.getNamespace())
+    );
+    startFields.add(
+        Fields.stringField(StoreDefinition.RepositorySourceControlMetadataStore.TYPE_FIELD,
+            type)
+    );
+
+    if (request.getFilter().getSyncStatus() != null) {
+      startFields.add(
+          Fields.booleanField(StoreDefinition.RepositorySourceControlMetadataStore.IS_SYNCED_FIELD,
```

Comment:
If we are using enums as API filters we might want to use the same for DB. Is there a specific reason why we decided upon filter to be an enum rather than boolean ?

Reply:
There's no specific reason for that, we could have boolean as well. I have made the change